src/collections/options.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright DataStax, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * deleteOneOptions
+ */
+export interface DeleteOneOptions {
+    sort?: Record<string, 1 | -1>;
+}
+
+export const deleteOneInternalOptionsKeys: Set<string> | null = null;
+
+/**
+ * findOptions
+ */
+export interface FindOptions {
+    limit?: number;
+    skip?: number;
+    sort?: Record<string, 1 | -1>;
+    projection?: Record<string, 1 | -1>;
+}
+
+class FindOptionsInternalClass {
```

Comment:
It seems a bit verbose to create an entire class just to get the interface keys. I'd recommend doing something like the following instead.

```ts
interface FindOneOptions {
  limit?: number;
  skip?: number;
  sort?: Record<string, 1 | -1>;
  projection?: Record<string, 1 | -1>;
}

const findOneOptionsKeys: Set<keyof Omit<FindOneOptions, 'sort'>> = new Set([
  'limit' as const,
  'skip' as const,
  'projection' as const,
  // Uncomment out the following line and TS will complain `Type '"sort"' is not assignable to type '"limit" | "skip" | "projection"'`
  // 'sort' as const
]);
```

Reply:
Cool, this is simple than creating a class I agree (`omit` trick didn't cross my mind when I was trying different options for this :-) ). I'll replace the classes with this in `options.ts` and ll update.

Only for `FindOptions` I require the equivalent internal interface, since I use that in `cursor.ts`, while constructing the internal options and also it has one extra field called `pageState`, so I'll have an interface just for that. 
For others, I just need the keys, I'll replace the class based solution with your suggestion above.

Reply:
Great :+1: worth noting though that TypeScript won't complain about missing entries in the set, so the following will compile fine:

```ts
interface FindOneOptions {
  limit?: number;
  skip?: number;
  sort?: Record<string, 1 | -1>;
  projection?: Record<string, 1 | -1>;
}

const findOneOptionsKeys: Set<keyof Omit<FindOneOptions, 'sort'>> = new Set([
  'limit' as const,
  'skip' as const,
  // Missing 'projection':
  // 'projection' as const,
]);
```

I don't know of a good way to fix this limitation, but I'm not sure it is worth the effort.